http_dumper_test.go

```diff
@@ -0,0 +1,190 @@
+package godump
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+)
+
+var ErrSimulatedNetwork = errors.New("simulated network error")
+
+func TestHTTPDebugTransport_WithDebugEnabled(t *testing.T) {
+	var buf bytes.Buffer
+
+	_ = os.Setenv("HTTP_DEBUG", "1")
+	defer os.Unsetenv("HTTP_DEBUG")
+
+	transport := NewHTTPDebugTransport(http.DefaultTransport)
+	transport.Dumper().writer = &buf
+	transport.SetDebug(true)
+
+	client := &http.Client{Transport: transport}
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("X-Test-Header", "TestValue")
+		w.WriteHeader(http.StatusOK)
+		if _, err := w.Write([]byte(`{"success":true}`)); err != nil {
+			t.Fatalf("failed to write response: %v", err)
+		}
+	}))
+	defer server.Close()
+
+	req, _ := http.NewRequestWithContext(context.Background(), http.MethodGet, server.URL, http.NoBody)
+	resp, err := client.Do(req)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	defer resp.Body.Close()
+
+	output := stripANSI(buf.String())
+	t.Logf("Captured dump:\n%s", output)
+
+	if !strings.Contains(output, "Transaction =>") {
+		t.Error("expected 'Transaction =>' in dump, got none")
+	}
+	if !strings.Contains(output, "Request =>") {
+		t.Error("expected 'Request =>' in dump, got none")
+	}
+	if !strings.Contains(output, "Response =>") {
+		t.Error("expected 'Response =>' in dump, got none")
+	}
+	if !strings.Contains(output, `"success":true`) {
+		t.Error("expected JSON body in dump")
+	}
+}
+
+func TestHTTPDebugTransport_WithDebugDisabled(t *testing.T) {
+	var buf bytes.Buffer
+
+	_ = os.Unsetenv("HTTP_DEBUG")
+
+	transport := NewHTTPDebugTransport(http.DefaultTransport)
+	transport.Dumper().writer = &buf
+
+	client := &http.Client{Transport: transport}
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	}))
+	defer server.Close()
+
+	req, _ := http.NewRequestWithContext(context.Background(), http.MethodGet, server.URL, http.NoBody)
```

Comment:
It doesn't make sense to me to use

NewRequestWithContext while passing a context.Background()

Use NewRequest directly